content/blog/multi-single-tenant-github-app-installation/index.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Multi Single Tenant Architecture with a GitHub App installations
+date: '2021-08-03T10:30:00.0Z'
+description: The story about how Roadie supports GitHub App installations.
+tags: ['github', 'backstage', 'architecture']
+author:
+  name: Brian Fletcher
+---
+
+The service we provide at Roadie (Backstage) heavily depends on the use of the GitHub APIs to provide it’s basic features. We discovered at some point in our journey that we could ask customers to provide us with a GitHub personal access token (PAT). However we discovered that companies are sensitive around providing PATs to third parties. As such we needed to use GitHub Apps.
+
+In this article, I will describe how we did it.
+
+# Background
+
+Roadie provides a SaaS version of the Backstage application that was open sourced by Spotify early last year. It is a single tenant application.
+
+In order to provide Backstage as a service we have settled on a multi single tenant architecture. This means that for each customer we provision a set of resources unique to the customers. We do have some shared resources between customers, however the main components of Backstage remain unique to each tenant. In the early stages we explored a purely multi tenanted application. However we felt that we would continue to fight against the tide of a single tenanted OpenSource Backstage project.
```

Comment:
"This means that for each customer we provision a set of resources unique to the customers."

to 

"This means that we provide unique resources for each customer we provision."

Reply:
Ive totally rewritten.